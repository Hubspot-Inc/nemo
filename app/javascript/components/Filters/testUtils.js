import FiltersModel from './model';
import ConditionSetModel from '../ConditionSetFormField/model';
import { submitterType } from './SubmitterFilter/component';

export const getFiltersStore = () => new FiltersModel({
  conditionSetStore: new ConditionSetModel({
    conditions: [
      {
        leftQingId: '1',
        op: 'eq',
        operatorOptions: [
          { name: '= equals', id: 'eq' },
          { name: 'â‰  does not equal', id: 'neq' },
        ],
        value: '5',
      },
    ],

    refableQings: [
      {
        id: '1',
        code: 'One',
        fullDottedRank: '1',
        rank: 1,
      },
      {
        id: '2',
        code: 'Two',
        fullDottedRank: '2',
        rank: 2,
      },
    ],
  }),

  allForms: [
    { id: '1', name: 'One' },
    { id: '2', name: 'Two' },
    { id: '3', name: "Q\"uo'te`s" },
  ],

  originalFormIds: [
    '1',
  ],

  selectedFormIds: [
    '2',
  ],

  originalIsReviewed: null,

  isReviewed: true,

  allSubmittersForType: {
    [submitterType.USER]: [
      { id: 'A', name: 'User A' },
      { id: 'B', name: 'User B' },
    ],
    [submitterType.GROUP]: [
      { id: 'A', name: 'Group A' },
      { id: 'B', name: 'Group B' },
    ],
  },

  originalSubmittersForType: {
    [submitterType.USER]: [{ id: 'A', name: 'User A' }],
    [submitterType.GROUP]: [],
  },

  selectedSubmittersForType: {
    [submitterType.USER]: [{ id: 'B', name: 'User B' }],
    [submitterType.GROUP]: [{ id: 'B', name: 'Group B' }],
  },

  advancedSearchText: 'query',
});
