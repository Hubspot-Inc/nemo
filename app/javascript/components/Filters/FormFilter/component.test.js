import React from 'react';
import { shallow } from 'enzyme';

import { getFiltersStore } from '../testUtils';

import Component from './component';

const defaultProps = {
  filtersStore: getFiltersStore(),
  onSubmit: jest.fn(),
};

it('renders as expected', () => {
  const wrapper = shallow(<Component {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});

describe('popover', () => {
  const wrapper = shallow(<Component {...defaultProps} />);
  wrapper.find('Button#form-filter').simulate('click');

  const overlay = shallow(wrapper.find('OverlayTrigger').prop('overlay'));

  it('handles callbacks', () => {
    overlay.find('Select2').simulate('change', { target: { value: defaultProps.filtersStore.allForms[0].id } });
    overlay.find('Button.btn-apply').simulate('click');

    expect(defaultProps.onSubmit).toMatchSnapshot();
  });
});
