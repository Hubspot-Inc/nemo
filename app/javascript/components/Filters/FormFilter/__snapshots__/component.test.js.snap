// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popover handles callbacks 1`] = `
[MockFunction] {
  "calls": Array [
    Array [],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`renders as expected 1`] = `
<OverlayTrigger
  containerPadding={25}
  id="form-filter"
  overlay={
    <Popover
      className="filters-popover"
      id="form-filter"
    >
      <Select2
        data={
          Array [
            Object {
              "id": "1",
              "text": "One",
            },
            Object {
              "id": "2",
              "text": "Two",
            },
            Object {
              "id": "3",
              "text": "Q\\"uo'te\`s",
            },
          ]
        }
        onSelect={[Function]}
        onUnselect={[Function]}
        options={
          Object {
            "allowClear": true,
            "dropdownCssClass": "filters-select2-dropdown",
            "placeholder": "Choose a form",
            "width": "100%",
          }
        }
        value="2"
      />
      <div
        className="btn-apply-container"
      >
        <Button
          className="btn-apply"
          onClick={[MockFunction]}
        >
          Apply
        </Button>
      </div>
    </Popover>
  }
  placement="bottom"
  rootClose={true}
  trigger="click"
>
  <Button
    id="form-filter"
    variant="secondary"
  >
    Form (One)
  </Button>
</OverlayTrigger>
`;
